src/Resizer.cs

```diff
@@ -1,27 +1,56 @@
-﻿using System;
+using System;
 using System.Collections.Generic;
 using System.IO;
-using ImageSharp;
+using SixLabors.ImageSharp;
+using SixLabors.ImageSharp.Formats;
+using SixLabors.ImageSharp.Formats.Bmp;
+using SixLabors.ImageSharp.Formats.Gif;
+using SixLabors.ImageSharp.Formats.Jpeg;
+using SixLabors.ImageSharp.Formats.Png;
+using SixLabors.Primitives;
 
 namespace ImageResizer
 {
-    public static class Resizer
+    public class Resizer
     {
-        public static void Resize(string filePath, IEnumerable<int> widths)
+        private Image<Rgba32> _image;
+        private IImageEncoder _encoder;
+
+        public void Resize(string filePath, IEnumerable<int> widths)
         {
             using (var stream = File.OpenRead(filePath))
             {
-                var image = Image.Load(stream);
-                var aspectRatio = new Size(image.Width, image.Height).AspectRatio();
+                _image = Image.Load(stream);
+
+                var filetype = Path.GetExtension(filePath);
+
+                switch (filetype)
+                {
+                    case ".jpg":
+                    case ".jpeg":
+                        _encoder = new JpegEncoder{Quality = 255};
+                        break;
+                    case ".gif":
+                        _encoder = new GifEncoder();
+                        break;
+                    case ".bmp":
+                        _encoder = new BmpEncoder();
+                        break;
+                    case ".png":
+                        _encoder = new PngEncoder();
+                        break;
+                }
+
+                var aspectRatio = new Size(_image.Width, _image.Height).AspectRatio();
                 foreach (var width in widths)
                 {
                     var size = new Size(width, (int)(width / aspectRatio));
-                    Resize(filePath, image, size);
+                    Resize(filePath, size);
                 }
             }
         }
 
-        private static void Resize(string filePath, Image image, Size newSize)
+        private void Resize(string filePath, Size newSize)
```

Comment:
let's keep the `image` being passed in so this method can remain `static`

Reply:
I removed `static` because it was listed as an item in the Issue.  I had to remove `Image` from the signature because SixLabors.ImageSharp.Image is static, and it wouldn't build because you "Cannot declare a variable of static type..."

I'll spend some more time looking at this, but there might be no other way around it.